packages/react-utilities/src/compose/types.ts

```diff
@@ -113,6 +113,34 @@ export type ComponentState<Shorthands extends ObjectShorthandPropsRecord> = {
   };
 } & Shorthands;
 
+/**
+ * This is part of a hack to infer the element type from a native element *props* type.
+ * The only place the original element is found in a native props type (at least that's workable
+ * for inference) is in the event handlers, so some of the helper types use this event handler
+ * name to infer the original element type.
+ *
+ * Notes:
+ * - Using an extremely obscure event handler reduces the likelihood that its signature will be
+ *   modified in any component's props.
+ * - Inferring based on a single prop name instead of a larger type like `DOMAttributes<T>` should be
+ *   less expensive for typescript to evaluate and is less likely to result in type expansion in .d.ts.
+ */
+type SomeEventName = 'onLostPointerCaptureCapture';
+
+/**
+ * Return type for `React.forwardRef`, including inference of the proper typing for the ref.
+ */
+export type ForwardRefComponent<Props> = SomeEventName extends keyof Props
+  ? Required<Props>[SomeEventName] extends React.PointerEventHandler<infer Element>
+    ? React.ForwardRefExoticComponent<Props & React.RefAttributes<Element>>
+    : never
+  : never;
+// A definition like this would also work, but typescript is more likely to unnecessarily expand
```

Comment:
Commented out code

Reply:
This was actually intentional (at least for the PR, possibly to leave it there) as an example of something else that someone might be inclined to try changing it to in the future and why it may not work as well.